services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -831,12 +831,11 @@ export class WorkflowService {
       }
     }
 
-    const updatedWorkflow = await this.updateContextById(
```

Comment:
This change is problematic, the behavior is changing from merge to override the whole doc + from atomic operation to non-atomic operation